### PR TITLE
Set ESlint rules more strict 🚑

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: eslint rules are too loose, errors are passing though.

## What I did
As described in review comments #904 linting is rather non-strict right now.
Though I'm personally of the opinion too strict linting can be harmful, I agree, we could make it a bit more strict to weed out some bad code we have and prevent bad come from getting pushed to master.

- [x] use eslint to lint all `.js` & `.jsx` files
- [x] use eslint to lint all `.json` files
- [x] sort all package.json using `sort-package-json` (single time).
- [x] set rules to a much much much stricter set
- [x] fix all js-files errors
- [x] fix all json-files errors 
- [x] use remark-lint to lint all `.md` files
- [x] use remark-lint-code-eslint to lint all code-samples inside markdown files
- [x] use prettier
- [x] fix all markdown-files errors
- [x] setup `lint-staged`
- [x] test `lint-staged`
- [x] fix any regressions
- [x] more covfefe

## How to test
run `npm run lint`

### CODE REVIEW

File: packages/addon-info/src/components/PropVal.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ function previewProp(val) {
   }
 
   if (!braceWrap) return content;
-  return <span>{{ content }}</span>;
+  return <span>{`{ ${content} }`}</span>;
```
Comment (subpopular) [inline packages/addon-info/src/components/PropVal.js]: nice!

**Comment by theinterned** [comment]
question: I've got this error:

```console
/Users/user/src/theinterned-storybook/packages/storybook-ui/src/modules/ui/components/left_panel/text_filter.test.js
  1:1  error  'enzyme' should be listed in the project's dependencies. Run 'npm i -S enzyme' to add it  import/no-extraneous-dependencies
```
this is due to enzyme being in the root `package.json` is it appropriate just add an `eslint-disable-line` or should we modify the eslint-config?

**Comment by theinterned** [comment]
@UsulPro has fixed some issues in the **addon-info** package in this PR: https://github.com/storybooks/storybook/pull/980

**Comment by theinterned** [comment]
I thought I'd give an update on my work. 

I'm down to `✖ 72 problems (64 errors, 8 warnings)`

For the time being I'm skipping jsx-a11y issues as I didn't feel comfortable fixing a number of the UI-related issues without running the "kitchen-sink" project and generally making more nuanced decisions. I plan to tackle that in a separate PR.

My PR is here https://github.com/storybooks/storybook/pull/1000

@ndelangen does it make sense to close that one in favour of this?

**Comment by ndelangen** [comment]
closed in favour of https://github.com/storybooks/storybook/pull/1000

**Comment by ndelangen** [comment]
Re-opening, will be closing #1000

**Review by shilman — APPROVED** [review]
@ndelangen this is looking much better. there is still some weirdness with the `prettier` config, but I couldn't figure out how to get rid of it:

```
/Users/shilman/projects/storybook/new/storybook/lib/ui/src/modules/ui/components/shortcuts_help.js
   69:53  warning  Delete `·(`  prettier/prettier
   75:3   warning  Delete `)`   prettier/prettier
   ...
```

Also when I run `npm run lint:js -- --fix` it removes a bunch of these parens and makes the code harder to read.

Finally, there are some warnings when I run the test suite.

Nevertheless, I think this is good to merge. What do you think?

File: .remarkrc [hunk 2]
```diff
@@ -5,7 +5,8 @@
     ["remark-lint-code", {"js": {
       "module": "node_modules/remark-lint-code-eslint",
       "options": {
-        "fix": true
+        "fix": true,
+        "configFile": "/Users/dev/Projects/GitHub/storybook/react-storybook/.eslintrc-markdown.js"
```
Comment (shilman) [inline .remarkrc]: @ndelangen ???

File: docs/pages/addons/api/index.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ See how we can use this:
 
 ```js
 // Register the addon with a unique name.
-addonAPI.register('kadira/notes', (storybookAPI) => {
+addonAPI.register('my-organisation/my-addon', (storybookAPI) => {
```
Comment (shilman) [inline docs/pages/addons/api/index.md]: @ndelangen can we put this in a different PR? i think it is unrelated to eslint?

**Review by shilman — CHANGES_REQUESTED** [review]
Hey this looks great but I think we should get it merged and then do refinements on the merged version. I think this PR is getting out of control?

File: .remarkrc (hunk 2 above)
Comment (ndelangen) [inline .remarkrc]: Will fix this

**Review by shilman — CHANGES_REQUESTED** [review]
@ndelangen I am seeing an error trying to run gatsby in the docs

```
npm run dev
```

Huge string of errors:
```
ERROR in ./~/css-loader!./~/postcss-loader/lib!./components/Docs/Content/style.css
Module build failed: Error: No PostCSS Config found in: /Users/shilman/projects/storybook/new/storybook/docs/components/Docs/Content
    at Error (native)
    at /Users/shilman/projects/storybook/new/storybook/docs/node_modules/postcss-load-config/index.js:51:26

 @ ./components/Docs/Content/style.css 4:14-130 18:2-22:4 19:20-136
...
```

**Comment by ndelangen** [comment]
It's fixed @shilman !

**Comment by ndelangen** [comment]
I'm ready to merge this, who can test locally and approve?

**Pull request merged**